.github/copilot-instructions.md

```diff
@@ -0,0 +1,57 @@
+- Main entry: `src/index.ts` → `src/cli/cli.ts` (command parsing/execution).
+- Commands are modular, organized under `src/m365/` by workload (e.g., `spo`, `aad`, `teams`).
+- Each command is a class extending `Command` (`src/Command.ts`).
+- Command metadata and options: see `src/cli/CommandInfo.ts`, `src/cli/CommandOptionInfo.ts`.
+- Command discovery/build: see `scripts/write-all-commands.js` (generates `allCommands.json`).
+
+## Key Patterns & Conventions
+- **Global options**: All commands accept `--output`, `--query`, `--debug`, `--verbose` (see `src/GlobalOptions.ts`).
+- **Authentication**: Managed via `src/Auth.ts`, supports multiple auth types (Device Code, Certificate, Secret, Managed Identity, etc.).
+- **HTTP requests**: Use the wrapper in `src/request.ts` (Axios-based, with logging/debug hooks).
+- **Telemetry**: Centralized in `src/telemetry.ts`.
+- **Validation**: Use Zod schemas (`zod` import, see `src/Command.ts`).
+- **Prompting**: Use utilities in `src/utils/prompt.ts` for interactive input.
+- **Config**: User settings/configuration via `configstore` (see `src/config.ts`).
+- **Completion**: Shell completion logic/scripts in `scripts/Register-CLIM365Completion.ps1` and `scripts/Test-CLIM365Completion.ps1`.
+
+## Developer Workflows
+- **Build**: `npm run build` (TypeScript compile + command metadata generation)
+- **Test**: `npm test` (runs version check, lint, and Mocha tests)
+- **Lint**: `npm run lint` (uses custom ESLint rules from `eslint-rules/`)
+- **Watch**: `npm run watch` (TypeScript in watch mode)
+- **Symlink for local CLI**: `npm link` (after build)
+- **Node version**: Must be 22 (see `scripts/check-version.js`)
+- **Docs**: Docusaurus site in `docs/` (config: `docs/docusaurus.config.ts`)
+
+## Project-Specific Notes
+- **Command structure**: Each command is a class, not a function. Use inheritance from `Command`.
+- **Command registration**: New commands must be discoverable for `write-all-commands.js` to pick up.
+- **SPFx support**: Special logic for SPFx project upgrades and compatibility checks in `src/m365/spfx/`.
+- **Output**: Prefer returning objects/arrays; formatting handled by CLI core.
+- **No direct file/console output in commands**: Use provided logger and output mechanisms.
+- **Testing**: Mocha-based, see test files alongside source (e.g., `*.spec.ts`).
+- **Custom ESLint rules**: See `eslint-rules/` and `eslint-plugin-cli-microsoft365` in `package.json`.
+
+## Integration & External Dependencies
+- **Microsoft Graph, SharePoint REST, etc.**: Use `request.ts` for all HTTP calls.
+- **Authentication**: Uses `@azure/msal-node` and related packages.
+- **Telemetry**: Application Insights via `applicationinsights` package.
+- **Docs**: Docusaurus, see `docs/` and `docs/docusaurus.config.ts`.
+
+## Examples
+- Add a new command: create a class in the appropriate `src/m365/<workload>/commands/` folder, extend `Command`, implement `commandAction`, register options, and ensure it is discoverable.
```

Comment:
I wonder if it would be more helpful to point out the 5 places/files that need to be added/updated when adding a new command

Reply:
Good idea